scripts/relup-base-vsns.escript

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env escript
+%% -*- mode: erlang; -*-
+
+-mode(compile).
+
+-define(RED, "\e[31m").
+-define(RESET, "\e[39m").
+
+main(["base-vsns", To0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    To = strip_pre_release(To0),
+    #{from_versions := Froms} = fetch_version(To, VsnMap),
+    lists:foreach(
+     fun(From) ->
+       io:format(user, "~s~n", [From])
+     end,
+     filter_froms(Froms)),
+    halt(0);
+main(["otp-vsn-for", Vsn0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    Vsn = strip_pre_release(Vsn0),
+    #{otp := OtpVsn} = fetch_version(Vsn, VsnMap),
+    io:format(user, "~s~n", [OtpVsn]),
+    halt(0);
+main(["insert-new-vsn", NewVsn0, BaseFromVsn0, OtpVsn0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    NewVsn = strip_pre_release(NewVsn0),
+    BaseFromVsn = strip_pre_release(BaseFromVsn0),
+    OtpVsn = list_to_binary(OtpVsn0),
+    case VsnMap of
+      #{NewVsn := _} ->
+          io:format(user, ?RED ++ "Version ~s already in DB!~n" ++ ?RESET, [NewVsn]),
```

Comment:
Nit: maybe it's easier to create a function "print_warning" that encapsulates the escape codes?

Reply:
:heavy_check_mark: